sap/xssec/security_context.py

```diff
@@ -30,6 +31,25 @@ def _check_config(config):
         _check_if_valid(item, 'config.{0}'.format(prop))
 
 
+def _create_cert_and_key_files(cert_content, key_content):
+    cert_file = tempfile.NamedTemporaryFile(mode='w', delete=False)
+    cert_file.write(cert_content)
+    cert_file.close()
+    key_file = None
+    if key_content is not None:
+        key_file = tempfile.NamedTemporaryFile(mode='w', delete=False)
+        key_file.write(key_content)
+        key_file.close()
+    return cert_file.name, key_file.name if key_file else None
+
+
+def _delete_cert_and_key_files(cert_file_name, key_file_name):
+    if cert_file_name:
+        unlink(cert_file_name)
+    if key_file_name:
+        unlink(key_file_name)
```

Comment:
If the first delete operation somehow fails, the second delete (or unlink) will not occur. Also if the file is not closed correctly, the delete operation will fail on windows (to my understanding of windows and NTFS semantics).

It might be beneficial to rewrite the `try... finally` clause calling this function to use a `with` block instead, since `NamedTemporaryFile` objects can be used as context managers. This obviates the need for exception handling and file deletion (if `delete=True` is passed to `NamedTemporaryFile(...)`)

Reply:
It's ok for the unlink to fail although it's very unlikely. And if it fails there is really nothing we can do about it unless terminating the container. In case the container is compromised, the attacker can get those from env vars anyway. So it's not a big concern to me. And we don't support Windows btw.
Using `with` block with NamedTemporaryFile is a good idea in most cases, but in this case, it makes the code more verbose.

Reply:
Out of curiosity, how do you know with certainty that the code is always run inside a container?

For security reasons, a security minded end user (developer using this library) could opt to unset env vars after application startup to avoid that - right?